src/gretel_trainer/benchmark/entrypoints.py

```diff
@@ -0,0 +1,137 @@
+from __future__ import annotations
+
+import logging
+import shutil
+from inspect import isclass
+from pathlib import Path
+from typing import Optional, Type, Union, cast
+
+import pandas as pd
+from gretel_client.config import get_session_config
+
+from gretel_trainer.benchmark.core import BenchmarkConfig, BenchmarkException, Dataset
+from gretel_trainer.benchmark.custom.datasets import CustomDataset
+from gretel_trainer.benchmark.custom.models import CustomModel
+from gretel_trainer.benchmark.gretel.datasets import GretelDataset
+from gretel_trainer.benchmark.gretel.models import GretelModel
+from gretel_trainer.benchmark.job_spec import JobSpec, model_name
+from gretel_trainer.benchmark.session import Session
+
+logger = logging.getLogger(__name__)
+
+DatasetTypes = Union[CustomDataset, GretelDataset]
+ModelTypes = Union[
+    CustomModel,
+    Type[CustomModel],
+    GretelModel,
+    Type[GretelModel],
+]
+
+
+def compare(
+    *,
+    datasets: list[DatasetTypes],
+    models: list[ModelTypes],
+    config: Optional[BenchmarkConfig] = None,
+) -> Session:
+    _verify_client_config()
+    config = config or BenchmarkConfig()
+
+    working_dir_cleanup = None
+    if not config.working_dir.exists():
+        # clean up working_dir if we fail to prepare jobs
+        working_dir_cleanup = lambda: shutil.rmtree(config.working_dir)
+
+    try:
+        config.working_dir.mkdir(exist_ok=True)
+        standardized_datasets = [
+            _standardize_dataset(dataset, config.working_dir) for dataset in datasets
+        ]
+
+        job_specs = []
+        for dataset in standardized_datasets:
+            for model in models:
+                job_specs.append(JobSpec(dataset=dataset, model=_create_model(model)))
+
+        _validate_compare(job_specs)
+    except Exception as e:
+        if working_dir_cleanup is not None:
+            working_dir_cleanup()
+        raise e
+
+    session = Session(jobs=job_specs, config=config)
+    return session.prepare().execute()
+
+
+def launch(
+    *,
+    jobs: list[tuple[DatasetTypes, ModelTypes]],
+    config: Optional[BenchmarkConfig] = None,
+) -> Session:
+    _verify_client_config()
+
+    config = config or BenchmarkConfig()
+
+    working_dir_cleanup = None
+    if not config.working_dir.exists():
+        # clean up working_dir if we fail to prepare jobs
+        working_dir_cleanup = lambda: shutil.rmtree(config.working_dir)
+
+    try:
+        config.working_dir.mkdir(exist_ok=True)
+        job_specs = [
+            JobSpec(
+                dataset=_standardize_dataset(dataset, config.working_dir),
+                model=_create_model(model),
+            )
+            for dataset, model in jobs
+        ]
+    except Exception as e:
+        if working_dir_cleanup is not None:
+            working_dir_cleanup()
+        raise e
```

Comment:
@mikeknep based on tests in `test_bad_setup.py` I arrived with this, my thinking was this:
- how test describes this case is "if we weren't able to run anything, i.e. validation failed or client is not configured, we don't want to create the working dir"
- I think that working dir not existing was a proxy for checking that we didn't do any work, I wasn't sure if that dir not existing was an important part

At the end, as this is right now, the contract for that working dir is this:
- if it existed, we leave it, even with new files that we might have written as part of `_standarize_dataset`
- if it didn't exists, we will remove it if we get an exception before any work was done (Q: should this include failures in `prepare()`?)
- regardless if it existed or not, we will leave it if we started some work, the thinking here is that leaving these files may be useful for any troubleshooting/investigations.

One last someone related question: `Session` has `cleanup`, which removes the project that was created, should working dir be included in that cleanup?

Reply:
I don't think I had really ever thought this through so thoroughly :joy: I don't have a particularly strong opinion here, nor do I think we've documented/committed to any specific behavior. My main thought is that since this is often running on someone's local machine, we shouldn't delete anything that may have existed before we started working, so your first bullet point about the contract seems most important to me.

RE: `Session#cleanup`, I think that was primarily to assist with cleaning up _multiple_ projects, which happens when running with Trainer enabled—in that mode, each run gets its own project because datasets can be broken into smaller partitions and Trainer has its own artifact eviction strategy. I'm inclined to say that it should continue only doing that (i.e. a helper over gretel client SDK/API calls) and not do anything with the working directory—leave that to users explicitly doing an `rm -rf`. WDYT?

Reply:
Sounds good on the `Session#cleanup`.

And on the working dir itself - main reason I went into that rabbit hole was that the test was using that dir to assert that no work was done and I was thinking of using a different "signal" for that, as the working dir creation seemed like a low level detail that may not be the best thing to "inspect" in the test, but then started wondering if that dir's existence is actually part of the contract and took this into that direction :).

TL;DR for the handling as it is right now:
- if dir was there, leave it there with any contents that were there previously or created during the initial setup
- if dir wasn't there and we didn't get to do any work (which is indicated by running stuff on `Session`), then remove it, so it doesn't litter someone's disk.
- if dir wasn't there, but we did work, leave it.

Reply:
Seems reasonable to me 👍